internal/pkg/cli/app_upgrade.go

```diff
@@ -141,20 +142,20 @@ func (o *appUpgradeOpts) askName() error {
 	return nil
 }
 
-func shouldUpgradeApp(appName string, version string) bool {
-	diff := semver.Compare(version, deploy.LatestAppTemplateVersion)
+func shouldUpgradeApp(appName string, appVersion string) bool {
```

Comment:
Related to the question I had here https://github.com/aws/copilot-cli/pull/5021/files#r1239025625.

I think we should recommend running `app upgrade` only if the current app template is missing some latest features. I would expect that we return `true` here only in this case.

Basically, I think there are two types of "versioning":
1. The binary version used to deploy the app.
2. The version number on which the app template is updated.

I agree that we can use the binary version to version the app template, but we should:
1. Keep an internal records of 2
2. Let users be comfortable with having a "false-outdated" app template verion. That is, they should be comfortable when they see that the app template version is on v1.27.0, while the binary is v1.29.0. How to convey this message is the question.